File: HRMS/HRMS.Web/Modules/ServerTypes/Communication/SocialAccountColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SocialAccountRow } from "./SocialAccountRow";
import { SocialPlatform } from "./SocialPlatform";

export interface SocialAccountColumns {
    SocialAccountId: Column<SocialAccountRow>;
    Platform: Column<SocialAccountRow>;
    AccountName: Column<SocialAccountRow>;
    ProfileUrl: Column<SocialAccountRow>;
    ConnectedDate: Column<SocialAccountRow>;
    IsActive: Column<SocialAccountRow>;
}

export class SocialAccountColumns extends ColumnsBase<SocialAccountRow> {
    static readonly columnsKey = 'Communication.SocialAccount';
    static readonly Fields = fieldsProxy<SocialAccountColumns>();
}

[SocialPlatform]; // referenced types

File: HRMS/HRMS.Web/Modules/Communication/SocialAccount/SocialAccountColumns.cs
using Serenity.ComponentModel;

namespace HRMS.Communication.Columns;

[ColumnsScript("Communication.SocialAccount")]
[BasedOnRow(typeof(SocialAccountRow), CheckNames = true)]
public class SocialAccountColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int SocialAccountId { get; set; }
    public SocialPlatform Platform { get; set; }
    [EditLink]
    public string AccountName { get; set; }
    public string ProfileUrl { get; set; }
    public DateTime ConnectedDate { get; set; }
    public bool IsActive { get; set; }
}
